test/uk/gov/hmrc/agentsubscription/model/ValidatorSpec.scala

```diff
@@ -188,29 +338,226 @@ class ValidatorSpec extends UnitSpec {
+        validateOSName(".") shouldBe JsSuccess(".")
+      }
+
+      "contains forward slashes /" in {
+        validateOSName("/") shouldBe JsSuccess("/")
+      }
+
+      "1 or more characters" in {
+        validateOSName("1") shouldBe JsSuccess("1")
+      }
+
+      "40 or fewer characters" in {
+        validateOSName("1234567890123456789012345678901234567890") shouldBe JsSuccess("1234567890123456789012345678901234567890")
+      }
+    }
+
+    "reject name" when {
+      "contains ampersands &" in {
+        validateOSName("&") shouldBe nameValidationError
+      }
+
+      "contains single quotes '" in {
+        validateOSName("'") shouldBe nameValidationError
+      }
+
+      "contains backslashes \\" in {
+        validateOSName("\\") shouldBe nameValidationError
+      }
+
+      "contains other invalid symbols" in {
+        validateOSName("Agency;Co") shouldBe nameValidationError
+        validateOSName("#1 Agency Worldwide") shouldBe nameValidationError
+        validateOSName("|Agency|") shouldBe nameValidationError
+      }
+
+      "more than 40 characters" in {
+        validateOSName("12345678901234567890123456789012345678901") shouldBe nameValidationError
+      }
+
+      "contains only whitespace" in {
+        validateOSName("     ") shouldBe whitespaceValidationError
+      }
+
+      "string is empty but field is present" in {
+        validateOSName("") shouldBe whitespaceValidationError
+      }
+    }
   }
 
-  private def validateAddress(address: String) = {
-    addressValidation.reads(JsString(address))
+  "overseasEmailValidation" should {
+    def validateOSEmail(email: String) = overseasEmailValidation.reads(JsString(email))
+    def emailValidationError = JsError(ValidationError("error.email.invalid"))
+
+    "accept name" when {
+      "simple email address" in {
+        validateOSEmail("test@example.com") shouldBe JsSuccess("test@example.com")
+      }
+
+      "email is minimal" in {
+        validateOSEmail("a@b") shouldBe JsSuccess("a@b")
+      }
+
+      "contains lowercase characters" in {
+        validateOSEmail("abcdefghijklmnopqrstuvwxyz@c.d") shouldBe JsSuccess("abcdefghijklmnopqrstuvwxyz@c.d")
+      }
+
+      "contains uppercase characters" in {
+        validateOSEmail("ABCDEFGHIJKLMNOPQRSTUVWXYZ@c.d") shouldBe JsSuccess("ABCDEFGHIJKLMNOPQRSTUVWXYZ@c.d")
+      }
+
+      "contains numbers" in {
+        validateOSEmail("123@c.d") shouldBe JsSuccess("123@c.d")
+      }
+
+      "local-part contains hyphens" in {
+        validateOSEmail("a-b@c.d") shouldBe JsSuccess("a-b@c.d")
+      }
+
+      "local-part contains full stops" in {
+        validateOSEmail("a.b@c.d") shouldBe JsSuccess("a.b@c.d")
+      }
+
+      "local-part contains +" in {
+        validateOSEmail("a+b@c.d") shouldBe JsSuccess("a+b@c.d")
+      }
+
+      "local-part contains #" in {
+        validateOSEmail("a#b@c.d") shouldBe JsSuccess("a#b@c.d")
+      }
+    }
+
+    "reject name" when {
```

Comment:
is a@b acceptable? might as well check for when there is no ".com" for example

Reply:
This is standard email regex that we use elsewhere.
I've made the example test data more email-like though.